src/depends/Sha3/Sha3.c

```diff
@@ -298,7 +298,7 @@ void Keccak(unsigned int rate, unsigned int capacity, const unsigned char *input
 
     /* === Absorb all the input blocks === */
     while(inputByteLen > 0) {
-        blockSize = MIN(inputByteLen, rateInBytes);
+        blockSize = (unsigned)MIN(inputByteLen, rateInBytes);
```

Comment:
:memo: Let me know if the following alternative form is preferred:

```c
blockSize = MIN((unsigned)inputByteLen, rateInBytes);
```

The alternative form has the advantage that it guarantees (in the form of a compiler warning on some platforms) that `rateInBytes` will not be out of range, but it may be less readable for some people.

Reply:
➡️ Ah, the downside of the alternative form is it could cause an incorrect result if `inputByteLen` overflows.

Reply:
Not totally sure at the moment, but for our purposes I do not think this method is expected to be used with large values of inputByteLen

Reply:
I agree, but given the unknown I prefer the approach that is known to give a correct answer. If this was a C++ file we could use `numeric_cast`, but the approach in this PR should work well in context.

Reply:
Agree. In that case, we can stick to your current change.